apm-agent-core/src/main/java/co/elastic/apm/agent/bci/ElasticApmAgent.java

```diff
@@ -257,8 +259,10 @@ private static synchronized void initInstrumentation(final ElasticApmTracer trac
         Runtime.getRuntime().addShutdownHook(new Thread(ThreadUtils.addElasticApmThreadPrefix("init-instrumentation-shutdown-hook")) {
             @Override
             public void run() {
+                logger.info("Used instrumentation groups: {}", InstrumentationUsageUtil.getUsedInstrumentationGroups());
                 tracer.stop();
                 matcherTimers.clear();
+                LoggingConfiguration.shutdown();
```

Comment:
Do this through `co.elastic.apm.agent.context.LifecycleListener#stop`

Reply:
Done